### PR TITLE
Allow `ALTER CONNECTION` Updates for `resource_connection_postgres`

### LINKED ISSUE #513: Allow `ALTER CONNECTION` Updates for `resource_connection_postgres`

### PR DESCRIPTION
**Pull request by bobbyiliev**
Implementing the `ALTER CONNECTION` Updates for `resource_connection_postgres` as per #368

This also lays down the groundwork for the other connection resources as well.

Closes #513

### CODE REVIEW

File: pkg/resources/schema.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func ValidateConnectionSchema() *schema.Schema {
 	}
 }
 
-func IdentifierSchema(elem, description string, required bool) *schema.Schema {
+func IdentifierSchema(elem, description string, required, forceNew bool) *schema.Schema {
```
Comment (RobinClowers) [inline pkg/resources/schema.go:93]: Having two boolean values in a function signature is pretty confusing, it seems easy to confuse the order. Normally I'd vote for using a struct here, but I realize that there are a lot of calls to this function and it's main purpose is to build a struct in the first place, so I'm not sure what the best solution is.

File: pkg/provider/acceptance_connection_postgres_test.go [hunk 2]
```diff
@@ -113,6 +113,62 @@ func TestAccConnPostgres_disappears(t *testing.T) {
 	})
 }
 
+func TestAccConnPostgres_updateConnectionAttributes(t *testing.T) {
+	initialSecretName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	updatedSecretName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
+	connectionName := acctest.RandStringFromCharSet(10, acctest.CharSetAlpha)
```
Comment (RobinClowers) [inline pkg/provider/acceptance_connection_postgres_test.go:119]: Probably `initialConnectionName` for consistency.

**Review by RobinClowers — APPROVED** [review]

File: pkg/resources/schema.go (hunk 1 above)
Comment (bobbyiliev) [reply pkg/resources/schema.go]: Good idea! I just updated this to introduce a `IdentifierSchemaParams` struct.

Let me know what you think.

**Review by RobinClowers — APPROVED** [review]
Nice!

**Pull request merged**